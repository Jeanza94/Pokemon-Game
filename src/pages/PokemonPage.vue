<template>

    <h1 v-if="!pokemon">Espere por favor.....</h1>

    <div v-else>
        <h1>¿Quién es este pokemón?</h1>

        <!-- todo picture -->
        <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
        <!-- todo opciones -->
        <PokemonOptions 
            :pokemons="pokemonArr"
            @selection="checkAnswer"
        />
    </div>

    <template v-if="showAnswer">
        <h2>{{message}}</h2>
        <button @click="nuevoJuego">Nuevo juego</button>
    </template>
</template>

<script>

    import PokemonOptions from '@/components/PokemonOptions.vue';
    import PokemonPicture from '@/components/PokemonPicture.vue';
    
    import getPokemonOptions from '@/helpers/getPokemonOptions'


    export default {
        components: {
            PokemonOptions,
            PokemonPicture,
        },

        data() {
            return {
                pokemonArr: [],
                pokemon: null,
                showPokemon: false,
                showAnswer: false,
                message: '',
            }
        },

        methods: {
            async mixPokemonArray() {
                this.pokemonArr = await getPokemonOptions();
                
                const rndInt = Math.floor(Math.random() * 4);
                this.pokemon = this.pokemonArr[rndInt];
            },
            checkAnswer(pokemonId) {
                this.showPokemon = true;
                this.message = pokemonId === this.pokemon.id 
                                    ? 'Felicidades ese es el pokemon'
                                    : `Oops, el pokemon es ${this.pokemon.name}` 
                this.showAnswer = true;
            },
            nuevoJuego() {
                this.mixPokemonArray();
                this.showAnswer = false;
                this.showPokemon = false;
                this.pokemon = null;
            }
        },

        mounted() {
            this.mixPokemonArray();
        }
    }
</script>

