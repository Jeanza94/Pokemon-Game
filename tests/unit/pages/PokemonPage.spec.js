import { mount, shallowMount } from "@vue/test-utils";
import PokemonPage from '@/pages/PokemonPage'
import { pockemonMock } from "../mocks/pokemons.mock";

describe('Pruebas en la pagina PokemonPage', () => { 

    let wrapper;
    

    beforeEach(() => {
        wrapper = shallowMount(PokemonPage);
    })

    //esta prueba no ayuda mucho en este caso
    test('debe de hacer match con el snapshot ', () => { 
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('debe de llamar la funcion mixPokemonArray al montar', () => { 
        const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        const wrapper = shallowMount(PokemonPage);
        
        expect(mixPokemonArraySpy).toHaveBeenCalled()
    });

    test('debe de hacer match con el snapshot cuando carga la informacion', () => { 

        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pockemonMock,
                    pokemon: pockemonMock[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: '',
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot();

    });

    test('debe de mostrar los componentes de PokemonPicture y PokemonOptions', () => { 

        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pockemonMock,
                    pokemon: pockemonMock[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: '',
                }
            }
        });

        const pokemonPicture = wrapper.find('pokemon-picture-stub');
        const pokemonOptions = wrapper.find('pokemon-options-stub');
        //debe de existir los 2
        
        expect(pokemonPicture.exists()).toBeTruthy();
        expect(pokemonOptions.exists()).toBeTruthy();

        //pokemon picture atributo  pokemonId===1
        const { pokemonid, showPokemon } = pokemonPicture.attributes();
        expect( pokemonid ).toBe('1');
        //pokemonOptions attibute pokemons que exista
        const { pokemons } = pokemonOptions.attributes();
        expect(pokemons).not.toBe(undefined)
        
    });

    test('pruebas en la funcion checkAnswer', async() => { 

        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pockemonMock,
                    pokemon: pockemonMock[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: '',
                }
            }
        });

        await wrapper.vm.checkAnswer(1);
        expect(wrapper.find('h2').exists()).toBeTruthy();
        expect(wrapper.vm.showPokemon).toBeTruthy();
        expect(wrapper.find('h2').text()).toBe('Felicidades ese es el pokemon')

        await wrapper.vm.checkAnswer(10);
        expect(wrapper.vm.message).toBe(`Oops, el pokemon es ${pockemonMock[0].name}`)

    });

});