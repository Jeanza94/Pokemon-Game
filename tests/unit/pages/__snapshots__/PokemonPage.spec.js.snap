// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la pagina PokemonPage debe de hacer match con el snapshot  1`] = `
<h1>Espere por favor.....</h1>
<!--v-if-->
`;

exports[`Pruebas en la pagina PokemonPage debe de hacer match con el snapshot cuando carga la informacion 1`] = `
<div>
  <h1>¿Quién es este pokemón?</h1><!-- todo picture -->
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub><!-- todo opciones -->
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
</div>
<!--v-if-->
`;
